File: #03/Categoria 04/q10 - uniao/uniao.hs

uniao :: Eq a => [a] -> [a] -> [a]
uniao [] ys = ys
uniao xs [] = xs
uniao xs (y:ys)
    | y `notElem` xs = uniao xs (reverse ys) ++ [y]
    | otherwise = uniao xs (reverse ys)



main :: IO ()
main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO [Int]
    print $ uniao a b

File: #03/Categoria 04/q10 - uniao/uniao.js
uniao = (arr1,arr2) => Array.from(new Set([...arr1,...arr2]));

console.log(uniao([4,5], [1]));
console.log(uniao([4,5], [4,2,5]));
console.log(uniao([1,2,3], [2,4,6]));